src/librbd/internal.cc

```diff
@@ -1747,6 +1762,10 @@ namespace librbd {
       }
       cls_client::snapshot_add(&op, snap_id, snap_name);
       r = ictx->md_ctx.operate(ictx->header_oid, &op);
+
+      if (r == 0 && ictx->object_map != NULL) {
+        ictx->object_map->snapshot(snap_id);
```

Comment:
To make this consistent with the snapshot, we'll need to do the object map snapshot when we start using the new snapshot context. Currently it's only changed in ictx_refresh, once we get the notify from adding the snapshot. Since we've already done all the image owner checks here, it's probably simpler to take md_lock for write and add the new snap_id to ictx->snapc here, while we update the object map under the same lock.